src/buffer/mod.rs

```diff
@@ -1654,6 +1654,25 @@ impl TextBuffer {
                             };
                             // Our manually constructed UTF8 is never going to be invalid. Trust.
                             line.push_str(unsafe { str::from_utf8_unchecked(&visualizer_buf) });
+
+                            let visualizer_rect = {
+                                let cursor_visualizer = self.cursor_move_to_offset_internal(
+                                    cursor_beg,
+                                    global_off + chunk_off - 1,
+                                );
```

Comment:
This is accidentally quadratic, because it always starts from `cursor_beg` again. I'd do either of these two things:
* We split the while loop into two, where one _only_ handles tabs and the other handles all other control chars. Then we only do 2 (?) `cursor_move_to_offset_internal` calls to get the start and end of the control char range and highlight that entire block.
* We copy `cursor_beg` into a `mut cursor_visualizer` variable above and then use it to iterate over these chars via `cursor_visualizer = self.cursor_move_to_offset_internal(cursor_visualizer, ...)`.

I think the first option would be more ideal, but I think we can just go with the 2nd option for now.

Reply:
Therefore, I did the second!